bot/exts/evergreen/tic_tac_toe.py

```diff
@@ -0,0 +1,322 @@
+        await confirm_message.add_reaction(Emojis.decline)
+
+        def confirm_check(reaction: discord.Reaction, user: discord.User) -> bool:
+            """Check is user who reacted from who this was requested, message is confirmation and emoji is valid."""
+            return (
+                reaction.emoji in (Emojis.confirmation, Emojis.decline)
+                and reaction.message.id == confirm_message.id
+                and user == self.players[1].user
+            )
+
+        try:
+            reaction, user = await self.ctx.bot.wait_for(
+                "reaction_add",
+                timeout=60.0,
+                check=confirm_check
+            )
+        except asyncio.TimeoutError:
+            self.over = True
+            self.canceled = True
+            await confirm_message.delete()
+            return False, "Running out of time... Cancelled game."
+
+        await confirm_message.delete()
+        if reaction.emoji == Emojis.confirmation:
+            return True, None
+        else:
+            self.over = True
+            self.canceled = True
+            return False, "User declined"
+
+    async def add_reactions(self, msg: discord.Message) -> None:
+        """Add number emojis to message."""
+        for nr in Emojis.number_emojis.values():
+            await msg.add_reaction(nr)
+
+    def format_board(self) -> str:
+        """Get formatted tic-tac-toe board for message."""
+        board = list(self.board.values())
+        return "\n".join(
+            (f"{board[line]} {board[line + 1]} {board[line + 2]}" for line in range(0, len(board), 3))
+        )
+
+    async def play(self) -> None:
+        """Start and handle game."""
+        await self.ctx.send("It's time for game! Let's begin.")
+        board = await self.ctx.send(
+            embed=discord.Embed(description=self.format_board())
+        )
+        await self.add_reactions(board)
+
+        for _ in range(9):
+            if isinstance(self.current, Player):
+                announce = await self.ctx.send(f"{self.current.user.mention}, your turn! React to emoji to mark field.")
+            timeout, pos = await self.current.get_move(self.board, board)
+            if isinstance(self.current, Player):
+                await announce.delete()
+            if timeout:
+                await self.ctx.send(f"{self.current.user.mention} ran out of time. Canceling game.")
+                self.over = True
+                self.canceled = True
+                return
+            self.board[pos] = self.current.symbol
+            await board.edit(
+                embed=discord.Embed(description=self.format_board())
+            )
+            await board.clear_reaction(Emojis.number_emojis[pos])
+            if check_win(self.board):
+                self.winner = self.current
+                self.loser = self.next
+                await self.ctx.send(
+                    f":tada: {self.current} is won this game! :tada:"
```

Comment:
```suggestion
                    f":tada: {self.current} has won this game! :tada:"
```

Reply:
Maybe `{user} won this game!` even better? Has doesn't sound well in this context.

Reply:
Yea, that sounds good too.